server/test/filters/RecordCookieSizeFilterTest.java

```diff
@@ -0,0 +1,88 @@
+package filters;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static play.test.Helpers.fakeRequest;
+
+import akka.stream.testkit.NoMaterializer$;
+import com.google.common.collect.ImmutableList;
+import io.prometheus.client.CollectorRegistry;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.Before;
+import org.junit.Test;
+import play.libs.streams.Accumulator;
+import play.mvc.EssentialAction;
+import play.mvc.Http;
+import play.mvc.Result;
+import play.mvc.Results;
+
+public class RecordCookieSizeFilterTest {
+  @Before
+  public void clearRegistry() {
+    CollectorRegistry.defaultRegistry.clear();
+  }
+
+  @Test
+  public void testCookieSizesAreRecorded() throws Exception {
+    // Run filter with several requests with various cookie sizes.
+    RecordCookieSizeFilter filter = new RecordCookieSizeFilter();
+    List<Integer> cookieSizes = ImmutableList.of(1000, 2000, 2000, 3000, 3500, 4000);
+    for (int cookieSize : cookieSizes) {
+      assertThat(runFilterWithCookieSize(filter, cookieSize).status()).isEqualTo(200);
+    }
+
+    var metricFamilySamples = CollectorRegistry.defaultRegistry.metricFamilySamples();
+    assertThat(metricFamilySamples.hasMoreElements()).isTrue();
+    var mfs = metricFamilySamples.nextElement();
```

Comment:
**nitpick:** Thoughts on renaming this to something clearer? Maybe metricElement?

Reply:
I agree this should be clearer.

I had some trouble with the names. On line 34, the type is `Enumeration<MetricFamilySamples>`. So on line 36, the type is `MetricFamilySamples`.

I'll rename the variables to match the types better.